addons/cloud_storage_azure/utils/cleanup_cloud_storage_azure.py

```diff
@@ -0,0 +1,130 @@
+import requests
```

Comment:
> These seem essentially unusable as users who may know _nothing_ about python development are expected to understand that they need to update a bunch of undocumented globals to unspecified values directly in the Odoo source after they've stumbled upon this. Not to mention the use of third party dependencies which they might not be aware of in any way.
> 

The use of the script has been preferred over a cron job which would perform the deletion automatically, mostly in case of a backup restore. If a backup is restored but the cloud storage data has been removed, well... too bad. Better have a script ran manually once in a while for garbage collection.

It's definitively not a tool which should be ran by a regular user, but rather by a sys admin.

Reply:
@nim-odoo I don't see how delegating the task to a sysadmin solves the problem.  If your sysadmin runs the garbage collector right after you deleted some attachment by accident, and then restores a backup because you asked for it, the file is lost.  Unless I missed something, to me this argument is not valid.

Also, this looks quite impractical for customers that host their instance by Odoo.sh, for instance.  Installing one of those modules requires that they have some sysadmin that regularly connects to Odoo.sh servers to launch the script.  Very small customers probably don't have a sysadmin, which may be a motivation for hosting Odoo...

Regarding the backup restore question, you could make the cron delete unused remote files after some delay, so that the remote files are guaranteed to be accessible for some time after the deletion of their corresponding attachment.

That being said, in any case the operation should not be initiated by a regular user.

Reply:
@rco-odoo 
The those cleanup scripts are designed to be executed locally in a non production server or locally
0. the script is network io bound. And Odoo is not designed for this.
1. They require different google service account / azure app registration with `LIST` and `DELETE` permission of cloud storage blobs. It would be safer that the administrator never upload these secrets.
2. These scripts are public, the administrator can directly copy and paste the code from github and execute it
3. The settings for cloud storage are removed in neutralize.sql, so I guess at least it won't be a problem for our suppport team 🤔. But I am not sure what is the backup and restore flow for odoo.sh. I guess when backup, usually developers can get the database dump and the backup filestore. But they cannot easily create a cloud storage backup snapshot. So it might be better to keep these files in the cloud.